firedrake/functionspaceimpl.py

```diff
@@ -818,3 +823,64 @@ def ComponentFunctionSpace(parent, component):
     new.component = component
     new.parent = parent
     return new
+
+
+class RealFunctionSpace(FunctionSpace):
+    """:class:`FunctionSpace` based on elements of family "Real". A
+    :class`RealFunctionSpace` only has a single global value for the
+    whole mesh.
+
+    This class should not be directly instantiated by users. Instead,
+    FunctionSpace objects will transform themselves into
+    :class:`RealFunctionSpace` objects as appropriate.
+
+    """
+
+    def __init__(self, mesh, element, name):
+
+        self.name = name
+        self._index = None
+        self._ufl_element = element
+        self.fiat_element = None
+        self._mesh = mesh
+        self.dim = 1
+        self.rank = 0
+
+        self._node_count = 1
+
+        self.comm = mesh.comm
+
+        """:class:`RealFunctionSpace` objects have no node set."""
+        self.node_set = None
+        """:class:`RealFunctionSpace` objects have no dof set."""
+        self.dof_dset = op2.GlobalDataSet(self.make_dat())
```

Comment:
If I make two `RealFunctionSpace` objects, they currently don't share a `dof_dset`.  This may be a problem.